src/mod_pam_unix.c

```diff
@@ -63,7 +62,7 @@ write_config_unix (pam_module_t *this, enum write_type op, FILE *fp)
 	opt_set->enable (opt_set, "use_first_pass", TRUE);
       break;
     case ACCOUNT:
-      if (with_krb5 || with_ldap || with_lum || with_winbind || with_sss || with_himmelblau)
+      if (with_krb5 || with_ldap || with_lum || with_winbind || with_sss)
 	fprintf (fp, "account\trequisite\tpam_unix.so\t");
```

Comment:
I could just add another condition here that sets pam_unix.so to `sufficient` when pam_himmelblau follows. I've instead placed pam_himmelblau above pam_unix in the account stack.

Reply:
I'm honestly not sure why this has been set to `requisite` anyway. It seems this will cause all these other modules to fail also. Perhaps the solution is just to change this line to `sufficient`?

Reply:
Quite simple: the other modules are called before pam_unix.so.

Reply:
This account stack is complete nonsense. It looks like it's being dumped to common-account in reverse order.

Reply:
Perhaps what is confusing me then is how pam_unix is responding to an entra id user on the account stack. pam_unix is failing when encountering entra id users (which it seems it should not!), so authentication is terminated because pam_unix says the user isn't authorized.
This is only an issue on the account stack.

Reply:
If I put pam_himmelblau earlier on the account stack, then this isn't a problem (because the himmelblau module runs first and replies `sufficient`).